### PR TITLE
Enable .net 452 - part 02

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #716

## Changes
- Add .NET 452 to OpenTelemetry.Exporter.Prometheus.csproj
- Add .NET 452 to OpenTelemetry.Exporter.ZPages.csproj

## Reference:
- https://referencesource.microsoft.com/#mscorlib/system/datetimeoffset.cs,52ee670254934854
- https://referencesource.microsoft.com/#mscorlib/system/datetimeoffset.cs,8f33340c07c4787e

### Checklist
- [x] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public API reviewed

The PR will automatically request reviews from [code owners](https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/CODEOWNERS#L15) and trigger CI build and tests.

### CODE REVIEW

**Comment by eddynaka** [comment]
Questions:
- we have some projects that doesn't target net46, for example `OpenTelemetry.Exporter.Jaeger`. Should we add net452 and net46 for those as well? Below the list of projects in the same condition:
  - OpenTelemetry.Exporter.Zipkin
  - OpenTelemetry.Extensions.Hosting
- we have some tests that just targets netcoreapp3.1. Should we include the other frameworks as well?

**Comment by CodeBlanch** [comment]
> Questions:
> 
> * we have some projects that doesn't target net46, for example `OpenTelemetry.Exporter.Jaeger`. Should we add net452 and net46 for those as well? Below the list of projects in the same condition:
>   
>   * OpenTelemetry.Exporter.Zipkin

My opinion would be... if we're supporting net452 in the instrumentation we should in the exporters too. Otherwise, how will people look at their data? 😄 The effort could be huge though especially in Jaeger with the Thrift stuff.

>   * OpenTelemetry.Extensions.Hosting

OpenTelemetry.Extensions.Hosting is specific to the .net core bootstrap pattern so I don't think we need the targets there?

> * we have some tests that just targets netcoreapp3.1. Should we include the other frameworks as well?

My vote would be to add them. But watch out netframework tests don't run on Linux. That's why `OpenTelemetry.Instrumentation.AspNet.Tests.Win` is named with ".Win", it will only be run on Windows job in CI. That works fine when a project is Windows-only. If we mix targets, that will have to be solved somehow.

**Comment by cijothomas** [comment]
Lets start with supporting net452 in api/sdk. If its non-trivial to add net452 support in zipkin, then lets skip it for now. Exporters outside of this repo have the ability to support net452 now. (there are microsoft internal exporters which would want to support net452)

.Hosting - Though its meant for Asp.Net Core applications, they can target the .net framework as well! So ideally we want to support netframework. But most likely net472 or above only.

Can we do the following avoid the win/linux issue? (This is what is done in Ot.instrumentation.dependencies.tests)
```
<TargetFrameworks>netcoreapp3.1</TargetFrameworks>
    <TargetFrameworks Condition="$(OS) == 'Windows_NT'">$(TargetFrameworks);net452;net461</TargetFrameworks>
```

**Comment by eddynaka** [comment]
@cijothomas , thanks for the advice! i will take a look!

For the reviewers, i have a question: so, net452 doesn't support some functions DateTimeOffset. With that in mind, i was creating in each project an Extension just to net452.

Problem: we need this in many projects and tests and i'm n-plicating the DataOffsetExtensions everywhere.

Should i:
- create a shared project with that?
- keep n-plicating?

**Comment by CodeBlanch** [comment]
There are a couple files in the OpenTelemetry project that are linked into others. Check out:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/ea70fece4ff42b3c8821d958ca458e03fb925b8f/src/OpenTelemetry.Exporter.Jaeger/OpenTelemetry.Exporter.Jaeger.csproj#L14
The idea there is to achieve re-use without needing another project (dependency to our users).

**Comment by eddynaka** [comment]
@CodeBlanch @cijothomas , I did the last changes that we talked. Can you review again? Thank you

File: src/OpenTelemetry.Exporter.Prometheus/OpenTelemetry.Exporter.Prometheus.csproj [hunk 1]
```diff
@@ -1,12 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net46;netstandard2.0;</TargetFrameworks>
+    <TargetFrameworks>net452;net46;netstandard2.0;</TargetFrameworks>
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Prometheus/OpenTelemetry.Exporter.Prometheus.csproj:3]: I start to wonder maybe we can extract this into a common props file? Maybe next PR 😃

**Pull request merged**